Services/Devices.cs

```diff
@@ -298,17 +308,23 @@ public async Task DeleteListAsync(IEnumerable<string> deviceIds)
             }
             catch (TooManyDevicesException error)
             {
-                this.log.Error("Failed to delete devices, the batch is too big", error);
+                var msg = "Failed to delete devices, the batch is too big";
+                this.log.Error(msg, error);
+                this.diagnosticsLogger.LogDiagnosticsData(SERVICE_ERROR_EVENT, $"{msg}: {error.Message}");
                 throw;
             }
             catch (IotHubCommunicationException error)
             {
-                this.log.Error("Failed to delete devices (IotHubCommunicationException)", () => new { error.InnerException, error });
+                var msg = "Failed to delete devices (IotHubCommunicationException)";
+                this.log.Error(msg, () => new { error.InnerException, error });
+                this.diagnosticsLogger.LogDiagnosticsData(SERVICE_ERROR_EVENT, $"{msg}: {error.Message}");
                 throw;
             }
             catch (Exception error)
             {
-                this.log.Error("Failed to delete devices", error);
+                var msg = "Failed to delete devices";
+                this.log.Error(msg, error);
+                this.diagnosticsLogger.LogDiagnosticsData(SERVICE_ERROR_EVENT, $"{msg}: {error.Message}"); ;
```

Comment:
this is an async method, is it intentional not to wait it?

I know it's a very different, but I would have used a different approach:
1. when an error occurs, give it to the logger
2. the logger keeps the events in memory, it doesn't send them immediately
3. the logger has a thread that every 2-3 seconds check if there are events to send, and send them

Do you think this could be done in future?